src/server/object_services/md_store.js

```diff
@@ -1110,20 +1110,20 @@ class MDStore {
         return buckets;
     }
 
+    /**
+     * Find deleted objects that were deleted before max_delete_time and are reclaimed.
+     *
+     * @param {number} max_delete_time - timestamp in milliseconds
+     * @param {number} limit
+     * @returns {Promise<nb.ID[]>}
+     */
     async find_deleted_objects(max_delete_time, limit) {
-        const objects = await this._objects.find({
-            deleted: {
-                $lt: new Date(max_delete_time),
-                $exists: true // This forces the index to be used
-            },
-        }, {
-            limit: Math.min(limit, 1000),
-            projection: {
-                _id: 1,
-                deleted: 1
-            }
-        });
-        return db_client.instance().uniq_ids(objects, '_id');
+        const query = `SELECT _id 
+        FROM ${this._objects.name}
+        WHERE (to_ts(data->>'deleted')<to_ts($1) and data ? 'deleted' and data ? 'reclaimed') 
+        LIMIT ${Math.min(limit || 0, 1000)};`;
+        const result = await this._objects.executeSQL(query, [new Date(max_delete_time).toISOString()]);
```

Comment:
The default limit should not be 0. Also, to make it more readable, calculate the limit outside of the string literal or use a default value for the argument. 


```suggestion
        const query_limit = limit || 1000;
        const query = `SELECT _id 
        FROM ${this._objects.name}
        WHERE (to_ts(data->>'deleted')<to_ts($1) and data ? 'deleted' and data ? 'reclaimed') 
        LIMIT ${query_limit};`;
        const result = await this._objects.executeSQL(query, [new Date(max_delete_time).toISOString()]);
```

Reply:
It's better to ensure the limit is a maximum of 1K... So I will use: 
```suggestion
        const query_limit = Math.min(limit || 1000, 1000);
```